### ISSUE TITLE
[doc] Add known issues documentation section

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#12665]
After some chat, we thought that it would be a good idea to have a known issues section on the documentation for 8.12 .

The proposed format would be an index, linking to the documentation of the specific feature. In particular, a few things in the table are:
- limitations of the new Search implementation [likely there will be some related to kind search]
- require doesn't backtrack grammars properly
  + https://github.com/coq/coq/issues/12575
  + https://github.com/coq/coq/issues/11793
  + https://github.com/coq/coq/issues/12658
- native compiler + OSX https://github.com/coq/coq/issues/11178
- abstract https://github.com/coq/coq/issues/9146
- sphinx?
- Exact [already in the manual]
- RAM issues? https://github.com/coq/coq/issues/8241
- Performance issues [such as quadratic behavior in some tactics]

cc: https://github.com/coq/coq/pull/947 , we may take inspiration from that.

**Comment by jfehrle** [comment]
Good idea.  It would be nice to have the links go the other way, from the documentation of the item to the description.  Readers may not always think to look in a known issues section.  That may be a bit more work, though.

**Comment by ejgallego** [comment]
@Zimmi48 proposal is indeed to have an index that links to the relevant section in the documentation. Personally I'd add a short description to each element in the index so the page can be read over quickly.

**Comment by ejgallego** [comment]
@ppedrot also points out the known issues with GCC 10 and OCaml < 4.07

**Comment by ppedrot** [comment]
That's https://github.com/coq/coq/issues/12940.
